includes/main.js

```diff
@@ -17,19 +17,67 @@ var pictures = [
 	'images/landscape-9.jpg',
 	'images/pexels-photo-132037.jpeg',
 	'images/pretty.jpg',
-];
+]; 
+
+
+
 
 function initiateApp(){
-	/*advanced: add jquery sortable call here to make the gallery able to be sorted
-		//on change, rebuild the images array into the new order
-	*/
-	makeGallery(pictures);
+	if (typeof(Storage) !== 'undefined' || typeof(Storage !== 'null')){
+
+		// localStorage.setItem("arraySort",JSON.stringify(pictures));
+	
+		stored = JSON.parse(localStorage.getItem("arraySort"));
+
+		console.log(stored);
+		console.log(typeof(stored));
+	}
+	else {
+		// var stored = pictures;
+	}
+		 /* advanced: add jquery sortable call here to make the gallery able to be sorted
+		 on change, rebuild the images array into the new order  */
+
 	addModalCloseHandler();
+
+	var sortableObj = { //?makes the .sortable obj that holds the events
+		start: makeGallery(stored),
+		update: findSortOrder
+	}
+	$('#gallery').sortable(sortableObj); //?makes the pictures sortable and updates the array on change 
+	//? makeGallery at the start to create the gallery. console logs the new pictures array after update
+
+	var theGalleryFigure = $('#gallery>figure');
+	$(theGalleryFigure).click(displayImage);
 }
+//? --------finds and logs the current sort order and re arranges pictures array ----------
+function findSortOrder(){
+	var theOrder = $('#gallery').sortable('toArray',{attribute: 'style'});
+	for (var currentImg = 0; currentImg<theOrder.length;currentImg++){
+		var valToChangeIndex = theOrder[currentImg].lastIndexOf('images/');
+		var changedVal = theOrder[currentImg].slice(valToChangeIndex);//todo finds images/ and removes it everything before it
+		valToChangeIndex = changedVal.lastIndexOf('")');
+		changedVal = changedVal.slice(0,valToChangeIndex);//todo removes everything after the file extension from ") and on
+		theOrder[currentImg] = changedVal; //todo changes the current value to the new string shich is just the images/xxx.xxx
+	}
+	pictures = theOrder; //todo sets the pictures array to the newly ordered array
+	localStorage.setItem("arraySort",JSON.stringify(pictures));
+}
+//todo ----------makes the gallary----------
 function makeGallery(imageArray){
+	var pSize = imageArray.length;
```

Comment:
pSize is a poor name.  What's p?  Yeah, the original array was pictures, but that's rather obtuse